htlcswitch/circuit.go

```diff
@@ -0,0 +1,158 @@
+package htlcswitch
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"github.com/go-errors/errors"
+	"github.com/roasbeef/btcd/wire"
+	"sync"
+)
+
+// CircuitKey uniquely identifies an active Sphinx (onion routing) circuit
+// between two open channels. Currently, the rHash of the HTLC which created
+// the circuit is used to uniquely identify each circuit.
+type CircuitKey [sha256.Size]byte
+
+func (k *CircuitKey) String() string {
+	return hex.EncodeToString(k[:])
+}
+
+// PaymentCircuit is used by HTLC switch service in order to determine
+// backward path for settle/cancel HTLC messages. A payment circuit is created
+// once a htlc manager forwards an HTLC add request. Channel points contained
+// within this message is used to identify the source/destination HTLC managers.
+//
+// NOTE: In current implementation of HTLC switch, the payment circuit might be
+// uniquely identified by payment hash but in future we implement the payment
+// fragmentation which makes possible for number of payments to have
+// identical payments hashes, but different source and destinations.
+//
+// For example if Alice(A) want to send 2BTC to Bob(B), then payment will be
+// split on two parts and node N3 will have circuit with the same payment hash,
+// and destination, but different source (N1,N2).
+//
+//	    	  1BTC    N1   1BTC
+//    	      + --------- o --------- +
+//      2BTC  |	                      |  2BTC
+// A o ------ o N0	           N3 o ------ o B
+//	      |		              |
+// 	      + --------- o --------- +
+//	         1BTC     N2   1BTC
+//
+type PaymentCircuit struct {
+	// PaymentHash used as uniq identifier of payment (not payment circuit)
+	PaymentHash CircuitKey
+
+	// Src is the channel id from which add HTLC request is came from and
+	// to which settle/cancel HTLC request will be returned back. Once the
+	// switch forwards the settle message to the source the circuit is
+	// considered to be completed.
+	Src *wire.OutPoint
+
+	// Dest is the channel id to which we propagate the HTLC add request
+	// and from which we are expecting to receive HTLC settle request back.
+	Dest *wire.OutPoint
+}
+
+// NewPaymentCircuit creates new payment circuit instance.
+func NewPaymentCircuit(src, dest *wire.OutPoint, key CircuitKey) *PaymentCircuit {
+	return &PaymentCircuit{
+		Src:         src,
+		Dest:        dest,
+		PaymentHash: key,
+	}
+}
+
+// ID returns unique id of payment circuit.
+func (a *PaymentCircuit) ID() string {
```

Comment:
It was intended for next PR, but you are right, for clarity we should remove any sign of next PR in this one.